pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -2324,6 +2330,21 @@ def _map_usage(
     )
 
 
+def _map_uploaded_file(uploaded_file: UploadedFile, _provider: Provider[Any]) -> str:
+    """Map an UploadedFile to a file ID understood by OpenAI-compatible APIs."""
+    file = uploaded_file.file
+    if isinstance(file, str):
+        return file
+    if isinstance(file, FileObject):
+        return file.id
+
+    file_id = getattr(file, 'id', None)
```

Comment:
I don't think we need to support arbitrary objects with an id; rather just the types allowed on the future `OpenAIUploadedFile`: `str` and `FileObject`